core/hotspot/rule.go

```diff
@@ -64,51 +64,90 @@ func (t MetricType) String() string {
 	//		the ControlBehavior decides the behavior of traffic shaping controller
-	MetricType MetricType `json:"metricType"`
+	MetricType MetricType `json:"MetricType"`
 	// ControlBehavior indicates the traffic shaping behaviour.
 	// ControlBehavior only takes effect when MetricType is QPS
-	ControlBehavior ControlBehavior `json:"controlBehavior"`
+	ControlBehavior ControlBehavior `json:"ControlBehavior"`
 	// ParamIndex is the index in context arguments slice.
 	// if ParamIndex is great than or equals to zero, ParamIndex means the <ParamIndex>-th parameter
 	// if ParamIndex is the negative, ParamIndex means the reversed <ParamIndex>-th parameter
-	ParamIndex int `json:"paramIndex"`
+	ParamIndex int `json:"ParamIndex"`
 	// ParamKey is the key in EntryContext.Input.Attachments map.
 	// ParamKey can be used as a supplement to ParamIndex to facilitate rules to quickly obtain parameter from a large number of parameters
 	// ParamKey is mutually exclusive with ParamIndex, ParamKey has the higher priority than ParamIndex
-	ParamKey string `json:"paramKey"`
+	ParamKey string `json:"ParamKey"`
 	// Threshold is the threshold to trigger rejection
-	Threshold int64 `json:"threshold"`
+	Threshold int64 `json:"Threshold"`
 	// MaxQueueingTimeMs only takes effect when ControlBehavior is Throttling and MetricType is QPS
-	MaxQueueingTimeMs int64 `json:"maxQueueingTimeMs"`
+	MaxQueueingTimeMs int64 `json:"MaxQueueingTimeMs"`
 	// BurstCount is the silent count
 	// BurstCount only takes effect when ControlBehavior is Reject and MetricType is QPS
-	BurstCount int64 `json:"burstCount"`
+	BurstCount int64 `json:"BurstCount"`
 	// DurationInSec is the time interval in statistic
 	// DurationInSec only takes effect when MetricType is QPS
-	DurationInSec int64 `json:"durationInSec"`
+	DurationInSec int64 `json:"DurationInSec"`
 	// ParamsMaxCapacity is the max capacity of cache statistic
-	ParamsMaxCapacity int64 `json:"paramsMaxCapacity"`
+	ParamsMaxCapacity int64 `json:"ParamsMaxCapacity"`
 	// SpecificItems indicates the special threshold for specific value
-	SpecificItems map[interface{}]int64 `json:"specificItems"`
+	SpecificItems map[interface{}]int64
+}
+
+type SpecificItem struct {
+	Key   string `json:"key"`
+	Value int64  `json:"value"`
+}
+
+func (t MetricType) MarshalJSON() ([]byte, error) {
+	return json.Marshal(t.String())
+}
+func (c ControlBehavior) MarshalJSON() ([]byte, error) {
+	return json.Marshal(c.String())
+}
+
+// Store the key-value pairs in specificItems in a temporary slice
+func convertSpecificItems(specificItems map[interface{}]int64) []*SpecificItem {
+	items := make([]*SpecificItem, 0, len(specificItems))
+	for k, v := range specificItems {
+		key, ok := k.(string)
+		if !ok {
+			key = fmt.Sprintf("%v", k)
+		}
+		item := &SpecificItem{
+			Key:   key,
+			Value: v,
+		}
+		items = append(items, item)
+	}
+	return items
 }
 
 func (r *Rule) String() string {
-	b, err := json.Marshal(r)
+	type tempRule Rule
```

Comment:
It might be better to use fmt.Sprintf() directly here

Reply:
Sorry, do you mean to use fmt.Sprintf to get the serialized string of the structure first, and then serialize json.Marshal?

Reply:
@binbin0325 Is the new way ok, please give me some advice thanks